services/cd-service/pkg/repository/repository.go

```diff
@@ -173,6 +175,66 @@ func openOrCreate(path string, storageBackend StorageBackend) (*git.Repository,
 	return repo2, err
 }
 
+func GetTags(cfg RepositoryConfig, repoName string, ctx context.Context) (tags []*api.TagData, err error) {
+	var empty StorageBackend
+	repo, err := openOrCreate(repoName, empty)
```

Comment:
please use the same storageBackend as the existing call to `openOrCreate`

Reply:
I wanted to use an empty one to skip those steps in openOrCreate to skip that step since I don't need that part of the function, but I can add it in